test/validation/api/classification/odp_classification_common.c

```diff
@@ -363,22 +363,22 @@ odp_packet_t create_packet(cls_packet_info_t pkt_info)
 	uint8_t dst_mac[] = CLS_DEFAULT_DMAC;
 
 	payload_len = sizeof(cls_test_packet_t) + pkt_info.len;
-	if (pkt_info.gtpu)
+	if (pkt_info.l4_type == CLS_PKT_L4_GTP)
 		payload_len += sizeof(odph_gtphdr_t);
 
 	seqno = odp_atomic_fetch_inc_u32(pkt_info.seq);
 
 	vlan_hdr_len = pkt_info.vlan ? ODPH_VLANHDR_LEN : 0;
 	vlan_hdr_len = pkt_info.vlan_qinq ? 2 * vlan_hdr_len : vlan_hdr_len;
 	l3_hdr_len = pkt_info.ipv6 ? ODPH_IPV6HDR_LEN : ODPH_IPV4HDR_LEN;
-	l4_hdr_len = pkt_info.udp ? ODPH_UDPHDR_LEN : ODPH_TCPHDR_LEN;
+	l4_hdr_len = pkt_info.l4_type == CLS_PKT_L4_UDP ? ODPH_UDPHDR_LEN : ODPH_TCPHDR_LEN;
 	eth_type = pkt_info.ipv6 ? ODPH_ETHTYPE_IPV6 : ODPH_ETHTYPE_IPV4;
-	next_hdr = pkt_info.udp ? ODPH_IPPROTO_UDP : ODPH_IPPROTO_TCP;
-	next_hdr = pkt_info.sctp ? ODPH_IPPROTO_SCTP : next_hdr;
-	next_hdr = pkt_info.icmp ? ODPH_IPPROTO_ICMPV4 : next_hdr;
-	if (pkt_info.gtpu)
+	next_hdr = pkt_info.l4_type == CLS_PKT_L4_UDP ? ODPH_IPPROTO_UDP : ODPH_IPPROTO_TCP;
+	next_hdr = pkt_info.l4_type == CLS_PKT_L4_SCTP ? ODPH_IPPROTO_SCTP : next_hdr;
+	next_hdr = pkt_info.l4_type == CLS_PKT_L4_ICMP ? ODPH_IPPROTO_ICMPV4 : next_hdr;
```

Comment:
This is not nice and hard to read. Could you change this to something cleaner like a switch statement? Of if the l4_type was directly the protocol value, you could just assign it here (but then you would also have to move the sequence number of all UDP packets to where it is for GTP or somehow special case GTP packet creation).

Reply:
Ack